apps/judicial-system/web/src/routes/Judge/Confirmation/Confirmation.tsx

```diff
@@ -18,29 +13,116 @@ import { TIME_FORMAT, formatDate } from '@island.is/judicial-system/formatters'
 import {
+  Case,
   CaseAppealDecision,
   CaseState,
   CaseTransition,
+  SignatureResponse,
+  TransitionCase,
 } from '@island.is/judicial-system/types'
-import * as api from '../../../api'
 import { userContext } from '@island.is/judicial-system-web/src/utils/userContext'
 import { useHistory } from 'react-router-dom'
 import { PageLayout } from '@island.is/judicial-system-web/src/shared-components/PageLayout/PageLayout'
 import PoliceRequestAccordionItem from '@island.is/judicial-system-web/src/shared-components/PoliceRequestAccordionItem/PoliceRequestAccordionItem'
 import * as style from './Confirmation.treat'
+import {
+  ConfirmSignatureQuery,
+  RequestSignatureMutation,
+  TransitionCaseMutation,
+} from '@island.is/judicial-system-web/src/graphql'
+import { useMutation, useQuery } from '@apollo/client'
+import { PendingSignature } from '@island.is/judicial-system-web/src/graphql/schema'
+
+interface SigningModalProps {
+  caseId: string
+  pendingSignature: PendingSignature
+  setModalVisible: React.Dispatch<React.SetStateAction<boolean>>
+}
+
+const SigningModal: React.FC<SigningModalProps> = (
+  props: SigningModalProps,
+) => {
+  const history = useHistory()
+  const [signatureResponse, setSignatureResponse] = useState<
+    SignatureResponse
+  >()
+
+  const { data } = useQuery(ConfirmSignatureQuery, {
+    variables: {
+      id: props.caseId,
+      documentToken: props.pendingSignature.documentToken,
+    },
+  })
+  const resSignatureResponse = data?.confirmSignature
+
+  useEffect(() => {
+    if (resSignatureResponse) {
+      setSignatureResponse(resSignatureResponse)
+    }
+  }, [resSignatureResponse, setSignatureResponse])
+
+  const renderContolCode = () => {
+    return (
+      <>
+        <Box marginBottom={2}>
+          <Text variant="h2" color="blue400">
+            {`Öryggistala: ${props.pendingSignature.controlCode}`}
+          </Text>
+        </Box>
+        <Text>
+          Þetta er ekki pin-númerið. Staðfestu aðeins innskráningu ef sama
+          öryggistala birtist í símanum þínum.
+        </Text>
+      </>
+    )
+  }
+
+  return (
+    <Modal
+      title={
+        !signatureResponse
+          ? 'Rafræn undirritun'
+          : signatureResponse.documentSigned
+          ? 'Úrskurður hefur verið staðfestur og undirritaður'
+          : signatureResponse.code === 7023 // User cancelled
+          ? 'Notandi hætti við undirritun'
+          : 'Undirritun tókst ekki'
+      }
+      text={
+        !signatureResponse
+          ? renderContolCode()
+          : signatureResponse.documentSigned
+          ? 'Tilkynning hefur verið send á ákæranda og dómara sem kvað upp úrskurð.'
+          : 'Vinsamlegast reynið aftur svo hægt sé að senda úrskurðinn með undirritun.'
+      }
+      secondaryButtonText={
+        !signatureResponse
+          ? null
+          : signatureResponse.documentSigned
+          ? 'Loka glugga'
+          : 'Loka og reyna aftur'
```

Comment:
Will this project have to support English as well?

Reply:
And what about content management?
Wondering if it's only supposed to support Icelandic, should the "translations" be stored in a json file so someone from content management can more easily update the texts.

Reply:
This project will not be in other languages. It is used by judges and prosecutors in the Icelandic judicial system.